### PR TITLE
PRSD-496: Search basis adr draft

### PR DESCRIPTION
**Pull request by AEPR**
Proposed ADR for choosing trigram search as the basis of searching through e.g. names, emails, phone numbers etc.

### CODE REVIEW

File: adrs/0020-exact-trigram-search.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+* No indexing
+* Add GIN Index
+* Add GiST Index
+
+## Decision Outcome
+
+No firm decision reached because major factors are unknown - decision to be finalised when there is a realistic test
+dataset to measure performance on.
+
+Interim decision of Concatenate data values, word similarity over threshold & N results with the highest word similarity
```
Comment (Travis-Softwire) [inline adrs/0020-exact-trigram-search.md:58]: Is there an obvious X, Y, Z for "If it's slow we should try X first", "If we're getting lots of false positives we should try Y first", "If we're excluding the desired result a lot we should try Z first"? If so worth capturing that here.

**Review by Travis-Softwire — APPROVED** [review]
One thought about capturing some more of what is currently in your brain, but otherwise LGTM. I think we can take 0019 as approved as well as that seems like a slam dunk.

File: adrs/0020-exact-trigram-search.md (hunk 1 above)
Comment (AEPR) [reply adrs/0020-exact-trigram-search.md:58]: I've added a section on this - are you happy for this to get merged in?

Comment (Travis-Softwire) [reply adrs/0020-exact-trigram-search.md:58]: LGTM 👍

**Pull request merged**